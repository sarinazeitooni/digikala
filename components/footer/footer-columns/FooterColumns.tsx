import React from 'react';
import style from './style/footer-columns.module.scss';
import messages from "./messages/messages";
import { v4 as uuidv4 } from 'uuid';
const FooterColumns = () => {
    return (
        <div className={style['footer-columns-container']}>
            {
                messages.columns.map((item) => {
                    return (
                        <div className={style['column']} key={uuidv4()}>
                            <div className={style['column-title']}>{item.title}</div>
                            <div className={style['column-item-container']}>
                                {item.items && item.items.map((text) => {
                                    return (
                                        <div key={text} className={style['column-item']}>{text}</div>
                                    )
                                })}
                            </div>
                            {item.social && <>
                                <div className={style['links-container']}>
                                    <div className={style['links']}>
                                        {messages.links.map((linkItem)=>{
                                            return(
                                                <span key={uuidv4()}>
                                                    {linkItem}
                                                </span>
                                            )
                                        })}
                                    </div>
                                </div>
                                <div className={style['column-title']}>{item.news && item.news}</div>
                                <div className={style['input-email-container']}>
                                    {item.email && <input placeholder={item.email} type='text' className={style['email-input']}/>}
                                    {item.submit &&
                                    <button className={style['submit-button']} type='submit'>{item.submit}</button>}
                                </div>
                            </>}
                        </div>
                    )
                })
            }
        </div>
    )
}
export default FooterColumns;