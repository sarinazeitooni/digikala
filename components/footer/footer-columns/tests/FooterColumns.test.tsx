import {shallow} from "enzyme";
import FooterColumns from "../FooterColumns";
describe("FooterColumns component" , ()=>{
    it("should render correctly" , ()=>{
        const container = shallow(<FooterColumns/>);
        expect(container).toMatchSnapshot();
    })
});