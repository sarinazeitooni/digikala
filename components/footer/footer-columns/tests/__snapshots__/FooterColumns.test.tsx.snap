// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterColumns component should render correctly 1`] = `ShallowWrapper {}`;
